### ISSUE TITLE
Gettiing OOM with golang 1.20 (6.1.9 linux)

### ISSUE THREAD

**Issue by chmouel** [golangci/golangci-lint#3538]
### Welcome

- [X] Yes, I'm using a binary release within 2 latest major releases. Only such installations are supported.
- [X] Yes, I've searched similar issues on GitHub and didn't find any.
- [X] Yes, I've included all information below (version, config, etc.).
- [X] Yes, I've tried with the standalone linter if available (e.g., gocritic, go vet, etc.). (https://golangci-lint.run/usage/linters/)

### Description of the problem

I am not sure on which linter cause the issue but i am getting a OOM when running on golang1.20 (updated this morning). The 

### Version of golangci-lint

<details>

```console
$ golangci-lint --version
golangci-lint has version 1.50.1 built from 8926a95f on 2022-10-22T10:50:47Z
```

</details>

### Configuration file

<details>

```console
$ cat .golangci.yml
run:
  build-tags:
    - e2e
  skip-dirs:
    - vendor
linters-settings:
  gocritic:
    disabled-checks:
      - unlambda
  errcheck:
    exclude: .errcheck.txt
  gofumpt:
    extra-rules: true
linters:
  enable:
    - asciicheck
    - bodyclose
    - depguard
    - dogsled
    - durationcheck
    - exhaustive
    - exportloopref
    - forbidigo
    - forcetypeassert
    - gochecknoinits
    - dupl
    - errcheck
    - errorlint
    - gofumpt
    - goimports
    - gosec
    - gocritic
    - misspell
    - gosec
    - gosimple
    - govet
    - importas
    - makezero
    - nakedret
    - predeclared
    - revive
    - staticcheck
    - stylecheck
    - unused
    - unparam
    - gochecknoinits
      # - goconst
      # - gocyclo
    # - goerr113
    # - gofmt
    - goheader
    - gomodguard
    - goprintffuncname
    - nilerr
    - noctx
      # - nolintlint (gofumpt conflict with it, create space all the time so :shrug:)
    - prealloc
    - promlinter
    # - rowserrcheck
    # - sqlclosecheck
    - tparallel
    #- unconvert
    # - wastedassign
    - whitespace
```

</details>

### Go environment

<details>

```console
$ go version && go env
go version go1.20 linux/amd64
```

</details>

### Verbose output of running

<details>

```console
$ golangci-lint cache clean
$ golangci-lint run -v
INFO [config_reader] Config search paths: [./ /home/chmouel/GIT/go/src/github.com/openshift-pipelines/pipelines-as-code /home/chmouel/GIT/go/src/github.com/openshift-pipelines /home/chmouel/GIT/go/src/github.com /home/chmouel/GIT/go/src /home/chmouel/GIT/go /home/chmouel/GIT /home/chmouel /home /] 
INFO [config_reader] Used config file .golangci.yml 
INFO [lintersdb] Active 40 linters: [asciicheck bodyclose depguard dogsled dupl durationcheck errcheck errorlint exhaustive exportloopref forbidigo forcetypeassert gochecknoinits gocritic gofumpt goheader goimports gomodguard goprintffuncname gosec gosimple govet importas ineffassign makezero misspell nakedret nilerr noctx prealloc predeclared promlinter revive staticcheck stylecheck tparallel typecheck unparam unused whitespace] 
INFO [loader] Using build tags: [e2e]             
INFO [loader] Go packages loading at mode 575 (files|name|imports|types_sizes|compiled_files|deps|exports_file) took 1.208595798s 
INFO [runner/filename_unadjuster] Pre-built 0 adjustments in 22.931017ms 
INFO [linters_context] importas settings found, but no aliases listed. List aliases under alias: key.
[killed process by oom killer]
``

</details>

### Code example or link to a public repository

<details>


not sure which code loop and grow in memory but that's the public repo https://github.com/openshift-pipelines/pipelines-as-code
```go
// add your code here
```

</details>

**Comment by ldez** [comment]
duplicate of #3536,#3535,#3533,#3470

**Comment by ajanata** [comment]
> duplicate of #3536,#3535,#3533,#3470

All of those duplicate issues are closed, however I am still experiencing this problem. Which issue is the active issue for tracking this?

**Comment by ldez** [comment]
As I wrote in the issues the problem has been fixed in #3414.

- golangci-lint <=v1.50.x is not compatible with go1.20.
- The latest release of golangci-lint ([v1.51.0](https://github.com/golangci/golangci-lint/releases/tag/v1.51.0)) is compatible with go1.20.

https://github.com/golangci/golangci-lint/blob/master/CHANGELOG.md#v1510

**Comment by chmouel** [comment]
confirmed as fixed for me thanks @ldez !
